src/getExtendedPublicKey.c

```diff
@@ -76,18 +48,16 @@ void handleGetExtendedPublicKey(
 
 	initializePath(dataBuffer);
 
-	derive_bip32_node_private_key();
+	derivePrivateKey(
+	        data.bip32Path,
+	        data.pathLength,
+	        data.chainCode,
+	        &data.privateKey
+	);
 
-	// We should do cx_ecfp_generate_pair here, but it does not work in SDK < 1.5.4,
-	// should work with the new SDK
-	cx_eddsa_get_public_key(
-	        &data.privateKey,
-	        CX_SHA512,
-	        &data.publicKey,
-	        NULL, 0, NULL, 0);
+	derivePublicKey(&data.privateKey, &data.publicKey);
 
 	os_memset(&data.privateKey, 0, sizeof(data.privateKey));
```

Comment:
Could we please wrap block starting from derivePrivateKey in FINALLY so memset is called regardless if something throws?